### ISSUE TITLE
Berufsbezeichnung

### ISSUE THREAD

**Issue by jwssnr** [jwssnr/feeef2#10]
Lasst uns Namen mögliche Namen suchen. Alles was passen könnte oder aus bestimmten gründen auch nicht passen könnte. Hier mal ein paar Vorschläge. (Ist mir aber klar, dass die Bezeichnung erst dann gesetzt werden kann, wenn die Positionierung klar ist.)

### Frontend-Entwickler
Weit verbreitet, wenn auch je nach Positionierung nicht ganz zutreffend … vielleicht.

### Fullstack-Entwickler
Geht wohl zu weit in die Applikationsentwicklung. Würde die Ausbildungszeit dazu überhaupt reichen?

### Theme-Entwickler
Ein bisschen zu nah an CMS aber sonst sehr passend.

### UI-Entwickler
Der Begriff UI ist nicht ganz klar. Oder schon?

**Comment by purzlbaum** [comment]
Wenn man sich die Entwicklung ansieht, waren früher alle Applikationsentwickler oder Programmierer. Im Laufe der Zeit hat aber eine Spezialisierung auf Front- oder Backend stattgefunden. Aktuell sehe ich beim Frontend-Entwickler eine weitere Aufsplittung in visuelle und technische Frontendler. Darum möchte ich hier zwei weitere Begriffsbezeichnungen ins Rennen werfen:

### Visual Frontend Developer (visueller Frontend-Entwickler)
- hat ein Flair für Design
- bewegt sich im HTML, CSS (LESS, SASS, SCSS) Umfeld

### Technical Frontend Developer (technischer Frontend-Entwickler)
- bewegt sich eher im JS Kosmos

### Fullstack Frontend Developer (visueller und technischer Frontend Entwickler)
- ist sowohl visual und technical Frontend Developer

Ähnlich wie bei den Friseuren könnte man nach dem Basislehrjahr wählen, in welche Richtung (Herren oder Damen, bzw visual oder technical) man gehen möchte. Nach Abschluss des einen oder anderen Zweigs könnte man mit einem Zusatzlehrjahr noch die andere Sparte lernen und sich dann Fullstack Frontend Developer nennen.

**Comment by dimitrisurber** [comment]
Wenn schon eine Frontend Ausbildung soll richtiges JS dabei sein. Mit HTML / CSS & bisschen Design füllst du doch keine 3 Jahre Ausbildung. Fokus wird am Ende sowieso von Lehrbetrieb gesetzt, nicht von der Schule oder dem Lehrplan, siehe Applikationsentwickler Ausbildung.

**Comment by jwssnr** [comment]
@purzlbaum – Sehr gut. Solche Begriffe habe ich gesucht. Mit dem Einwurf von @dimitrisurber bezüglich JS bin ich einverstanden. Ein **Visual Frontender** sollte soweit JS können, dass er damit UI-Dinge bauen oder einbauen kann, wie Toggles, Sliders, Maps, Scrollevents und solcherlei.

**Comment by jwssnr** [comment]
Für mein Verständnis erfüllen die Applikationsentwickler den Job der **Technical Frontender.** Diese Schublade könnte ich mir für Stellenprofile vorstellen, aber im Moment nicht für eine Grundbildung, denn das EFZ Applikationsentwickler ist (leider) als gegeben zu betrachten.

**Comment by signalwerk** [comment]
@purzlbaum 
Merci für deine Inputs! Cool!
Ich persönlich würde das Wort «Fullstack» nicht einbauen, da es sehr starkt mit Fontend zusammen mit Backend besetzt ist. Ich verstehe deine Überlegung, ist mir aber sehr nahe am gewohnten Begriff. «Fullrange», «360» oder so... Aber klingt dann hald bisschen nach Buzzword-Bingo.

**Comment by signalwerk** [comment]
@dimitrisurber total. die leute, die aus der ausbildung kommen sollen top notch JS dev's sein, denen ausser Erfahrung nichts fehlt. die sollen mit modernen JS-Frameworks schreiben können und ihr setup mit babel, webpack, blablablup einrichten.

**Comment by jwssnr** [comment]
Falls unser neuer Lehrberuf wirklich das **ganze Frontend** in der Grundbildung abdecken kann, bleiben wir sinniger Weise beim einfachen Begriff **Frontend-Entwickler** … falls. Ich finde die Inputs gut und die beiden Blickwinkel «Technical» und «Visual» merkwürdig im wörtlichen Sinne: Merken wir sie uns. Auch bin ich überzeugt, man darf hier nichts zusammenbasteln, was einerseits in der Praxis belächelt und umgangsprachlich zu fusselig ist. Ich sag nur «Polydesigner 3D». 😄

**Comment by signalwerk** [comment]
### User Interface Entwickler
Ich habe heute ein [Interview](https://devchat.tv/js-jabber/jsj-326-conversation-with-ember-co-creator-tom-dale-on-ember-3-0-and-the-future-of-ember/) gehört. Dabei wurde der Begrif «UI-Engeneer» (wohl UI-Entwickler in «Deutsch») für die Art von Personen verwendet, die wir wohl im Kopf haben. Wenn ich darüber nachdenke, so finde ich diesen Begriff eigentlich gar nicht schlecht, da wohl wesentlich mehr Leute das Wort UI (oder User Interface) verstehen statt «Frontend». Personen, die eher einen grafischen Hintergrund haben, können sich unter «Frontend» wohl nicht so viel vorstellen.

**Comment by jwssnr** [comment]
Ja, den hatte ich auch aufgeschnappt. Ich bin mit dem Begriff nicht ganz glücklich, weil User Interface schon älter ist als das Web selbst und deshalb natürlich vielerorts ausserhalb des Webs angewandt wird. In meinem Kontext kommt mir bei UI als erstes die Oberfläche meiner Tools in den Sinn. Und bei anderen vielleicht gar der Touchscreen der Kaffeemaschine. (Wobei natürlich auch da eigentlich Menschen mit Designkompetenzen dran arbeiten sollten.) Frontendmässig ist UI auch etwas wässerig. Gehört Mengentext zum UI oder nur die interaktiven Elemente?

**Comment by jwssnr** [comment]
# Entwickler?
Generell hadere ich immer mit dem Begriff Entwickler. Ist etwas hochgestochen nicht? Entwickeln wir wirklich? Oder bauen wir einfach? Ich nehme an, Frontend Developer ist angelehnt an Software Developer, aber ist das wirklich auf einem Level?

**Comment by signalwerk** [comment]
> Gehört Mengentext zum UI oder nur die interaktiven Elemente?

Eine berechtigte Frage, bei der die Bezeichnung in der Tat wässrig ist.

> Entwickler?

Ich persönlich verstehe mich als Entwickler, da ich mit meinem Kunde etwas entwickle und nicht einfach 1:1 umsetze. Dafür gibts günstigere Dienstleister im fernen Osten. Aber ich sehe natürlich auch deine Sichtweise

**Comment by jwssnr** [comment]
Natürlich. Wir entwickeln mit unseren Kunden zusammen ein Produkt. Hier bezieht sich das «entwickeln» aber auf den ganzen Prozess, von der Idee, dem Konzept, übers Design zur Realisation – ähnlich wie bei Designern. Der Schreiner wäre dann der Tisch-Entwickler.

Je länger sich die Gedanken um den Aufgabenbereich und die passende Berufsbezeichnung dreht, desto mehr komme ich wieder zurück auf den **Frontend-Entwickler**.
